src/cart/cart_service.bal

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/grpc;
+import ballerina/log;
+
+configurable string redisHost = "";
+configurable string redisPassword = "";
+listener grpc:Listener ep = new (9092);
+
+@grpc:ServiceDescriptor {descriptor: ROOT_DESCRIPTOR_DEMO, descMap: getDescriptorMapDemo()}
+service "CartService" on ep {
+    private final DataStore store;
+
+    function init() {
+        if redisHost == "" && redisPassword == "" {
```

Comment:
Rather than doing this lets have a configuration like `useInMemoryStore` and check that here.